src/Dialect/ONNX/Transforms/InstrumentONNXSignaturePass.cpp

```diff
@@ -71,30 +73,54 @@ class InstrumentONNXSignaturePass
 
   void runOnOperation() override {
     onnx_mlir::EnableByRegexOption traceSpecificOpPattern(
+
+        /*emptyIsNone*/ false);
+    onnx_mlir::EnableByRegexOption traceSpecificNodePattern(
         /*emptyIsNone*/ false);
+
     traceSpecificOpPattern.setRegexString(signaturePattern);
+    traceSpecificNodePattern.setRegexString(nodeNamePattern);
     // Iterate on the operations nested in this function.
     getOperation().walk([&](mlir::Operation *op) {
-      std::string opName = op->getName().getStringRef().str();
       auto dialect = op->getDialect();
+      Location loc = op->getLoc();
```

Comment:
Did not check the logic in great detail, I assume that you used a small example mlir file and tried both the signature and the new functionality and that you were happy with the outputs.